### ISSUE TITLE
Players cannot play music

### ISSUE THREAD

**Issue by Blacklands** [Matthew-Whitlock/FoundryVTT-music-permissions#2]
I'm trying this module for the first time. I'm on Foundry `9.249` (maybe that's already the issue? did something change in the latest version?).

I can modify the permissions of each playlists just fine. If I give a player the permissions to edit a playlist, they can do that just fine.
But no matter how I modify any of the settings, no player can play music (neither the whole playlist or individual songs). Yes, even if they're owners of it and everything is set up for that. They _always_ get

```
Error: You do not have permissions to play from this playlist!
```

Is that a known bug?
I've tried having the player create a whole new playlist, in case the existing ones don't work for some reason. Still doesn't work. I've restarted the world, changed player permissions and changed them back... no dice.

**Comment by pesky-wobbit** [comment]
Can't play sounds either. Would love to give my players this!

**Comment by Matthew-Whitlock** [comment]
This is not a known bug. The only locations in the code where that specific error is generated would point to this being an issue with permissions. Could you both check on the module setting for "Playback: " ? This sets a minimum ownership level that players should have of a playlist to be able to play from it.

If you believe you have this and the playlist permissions configured properly please let me know what settings you have currently. Thanks!

Edit: Please also let me know if you have any other music/sound-related modules currently

**Comment by pesky-wobbit** [comment]
I didn't notice the Playback setting! I fixated on "Minimum Control
Permission." Works as intended. Thanks!

**Comment by Blacklands** [comment]
I have the following module settings:
```
Playback: All players
Minimum control permission: Observer
```

- If I now set the permissions for a playlist to `All Players: Observer` and the specific user I want to be able to play songs I leave at `Default`, it works for that user.
- If I set `All Players: Observer` and the user also to `Observer`, I get the error message again and it doesn't work. That doesn't seem right...?
- For `All Players: None` and the user as `Observer`, I also get the error message.
- For `All Players: Limited` and the user as `Observer`, same behavior (error).
- For `All Players: Owner` and the user as `Default`, it again works fine and the user can play the playlist.
- For `All Players: Owner` and the user as `Observer`, I get the error message again.
- For `All Players: Observer` and the user as `None`, the user cannot see the playlist at all. (Makes sense to me.)

At this point I'm just very confused.
What is the relationship between the `All Players` setting and the individual settings, even? There's nothing about this on the Foundry website, it seems...

I would have expected that:

- The individual settings "inherit" the `All Players` setting in case of `Default`
- The individual settings override the `All Players` setting in all other cases (e.g. `All Players: Limited` and user: `Observer` would mean that the user's permission is now `Observer`)

But this doesn't seem to be the case at all. Why is `All Players: Observer` + user `Observer` something different (that doesn't work) from `All Players: Observer` + user `Default` (that _does_ work)?
But then lower permissions _do_ seem to override higher ones (individual `None`/`Limited` overrides `All Players: Observer`). But not the other way around...?

---

So this means that I can get the module to work if I set the permissions as `All Players: Observer` and then leave them at `Default` for the players who are supposed to have access to that playlist, and set them to `None` for everyone else. But that's not ideal, and I feel like I'm missing something here.

**Comment by Matthew-Whitlock** [comment]
This was an issue order-of-operations issue with the ?? operator in Javascript - thanks for the extra info on the bug and for reporting it here! It should be fixed in the new release 2.0.1 which I've just published. I'm going to close this, but feel free to reopen or open a new issue if this is still around for you.
